prov/efa/src/rdm/rxr_rma.c

```diff
@@ -322,6 +326,15 @@ ssize_t rxr_rma_post_write(struct rxr_ep *ep, struct rxr_op_entry *tx_entry)
 	if (peer->is_local && ep->use_shm_for_tx)
 		return rxr_rma_post_shm_write(ep, tx_entry);
 
+	if (!(tx_entry->fi_flags & FI_INJECT) && rxr_ep_determine_rdma_write_support(ep, tx_entry->addr, peer)) {
+		/* RDMA_WRITE does not support FI_INJECT, because device may
+		 * need to re-send data and FI_INJECT allows user to re-use
+		 * these buffers immediately
+		 */
+		/* TODO: do we ever use a SEND instead of a write based on total_len? */
+		rxr_op_entry_post_remote_write(tx_entry);
```

Comment:
return here.

Reply:
done, reworked some of this logic too to fix FI_REMOTE_CQ_DATA case too